packages/cmf/src/selectors/components.js

```diff
@@ -0,0 +1,41 @@
+import getToJSMemoized from './toJS';
+
+/**
+ * return a collection or subset of a collection from a cmf store
+ * this function returns collection from cmf.components, where the local state of some components
+ * @param {Object} state
+ * @param {String or Array<String>} collectionPath
+ * @example
+ *  get('foo.bar', true) === state.cmf.components.getIn(['foo', 'bar'], true)
+ */
+export function get(state, collectionPath, defaultValue) {
+	let path;
+	if (typeof collectionPath === 'string') {
+		path = collectionPath.split('.');
+	} else if (Array.isArray(collectionPath)) {
+		path = collectionPath;
+	}
+	if (path) {
+		return state.cmf.components.getIn(path, defaultValue);
+	}
+	throw Error(`Type mismatch: collectionPath should be a string or an array of string
+got ${collectionPath}`);
+}
+
+const selectors = {};
+
+export function toJS(state, path) {
+	const joinedPath = Array.isArray(path) ? path.join('.') : path;
+	if (!selectors[joinedPath]) {
+		selectors[joinedPath] = getToJSMemoized(calledState => get(calledState, path));
+	}
+	return selectors[joinedPath](state);
+}
+
```

Comment:
Functions "get" and "toJS" can be extracted as common, and can be used by collections and components, as they do pretty mutch the same

could be good to hear opinions, whether need to do this, or better leave like this